src/renderer/components/Settings/ProjectConfig.js

```diff
@@ -0,0 +1,130 @@
+// @ts-check
+import * as React from 'react'
+import { Button, makeStyles, Paper, Typography } from '@material-ui/core'
+import { defineMessages, useIntl } from 'react-intl'
+import { useQuery } from '@tanstack/react-query'
+import api from '../../new-api'
+import { ipcRenderer, remote } from 'electron'
+
+const m = defineMessages({
+  // title of project setting page
+  title: 'Project Settings',
+  projectName: 'Project Name',
+  config: 'Config',
+  importConfig: 'Import Config'
+})
+
+export const ProjectConfig = () => {
+  const classes = useStyles()
+  const { formatMessage: t } = useIntl()
+
+  /** @type {import('@tanstack/react-query').UseQueryResult<{name:string, version:string}|undefined>}  */
+  const { data: metadata } = useQuery(['metadata'], () => api.getMetadata())
+
+  /** @type {import('@tanstack/react-query').UseQueryResult<string>} */
+  const { data: encryptionKey } = useQuery(['encryptionKey'], () =>
+    api.getEncryptionKey()
+  )
+
+  async function importConfig () {
+    const result = await remote.dialog.showOpenDialog({
+      title: t(m.importConfig),
+      filters: [{ name: 'Mapeo Settings', extensions: ['mapeosettings'] }],
+      properties: ['openFile']
+    })
+    if (result.canceled) return
+    if (!result.filePaths || !result.filePaths.length) return
+
+    ipcRenderer.send('update-config', result.filePaths[0])
+    window.location.reload()
+    //   userConfig.importSettings(result.filePaths[0], err => {
+    //     if (err) return onerror(err)
+    //     ipc.send('reload-config', async err => {
```

Comment:
I need to call this IPC call. this IPC call originally was being sent from the main process to [mapeo-core](https://github.com/digidem/mapeo-desktop/blob/71276630bf09497529fb1fa1139261cf6beedba8/src/background/mapeo-core/index.js#L128), which is a background process.

Now we are sending this from a renderer process. Do you know the best way to do this? @gmaclennan 

Should i send an IPC call to the main thread, and then that IPC sends and ipc call to mapeo-core?

Reply:
@ErikSin we set up a MessageChannel to communicate between renderer processes. There is currently a message channel established between the [`mapeo-core`](https://github.com/digidem/mapeo-desktop/tree/master/src/background/mapeo-core) background process and the frontend. The message handlers are exported from [`mapeo-core/index.js`](https://github.com/digidem/mapeo-desktop/blob/master/src/background/mapeo-core/index.js), and the coordination between the message channel and the handlers is in [`src/background/index.js`](https://github.com/digidem/mapeo-desktop/blob/master/src/background/index.js). The initial connection of the MessageChannels needs to be coordinated via the main process, which is done here: https://github.com/digidem/mapeo-desktop/blob/master/src/background/index.js. Finally on the front-end renderer process the channel is connected in https://github.com/digidem/mapeo-desktop/blob/master/src/renderer/index-preload.js.

If you want to do IPC with mapeo-core, it is a question of adding extra handler methods to [`mapeo-core/index.js`](https://github.com/digidem/mapeo-desktop/blob/master/src/background/mapeo-core/index.js) and calling them from within https://github.com/digidem/mapeo-desktop/blob/master/src/renderer/new-api.js, which has access to the message channel.

If you want to do IPC to a different background renderer process (e.g. not mapeo-core), then you will need to set up a new channel, and attach it to `window` in the renderer. I can help with more details if that is what you need.